pkg/controllers/machine/disruption/drift_test.go

```diff
@@ -127,7 +127,142 @@ var _ = Describe("Drift", func() {
+			cp.Drifted = false
+			provisioner.Spec.Requirements = []v1.NodeSelectorRequirement{
+				{Key: v1alpha5.LabelCapacityType, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.CapacityTypeOnDemand}},
+				{Key: v1.LabelArchStable, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.ArchitectureAmd64}},
+				{Key: v1.LabelOSStable, Operator: v1.NodeSelectorOpIn, Values: []string{string(v1.Linux)}},
+			}
+			machine.Spec.Requirements = []v1.NodeSelectorRequirement{
+				{Key: v1alpha5.LabelCapacityType, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.CapacityTypeOnDemand}},
+				{Key: v1.LabelArchStable, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.ArchitectureAmd64}},
+				{Key: v1.LabelOSStable, Operator: v1.NodeSelectorOpIn, Values: []string{string(v1.Linux)}},
+			}
+
+			ExpectApplied(ctx, env.Client, provisioner, machine)
+			ExpectReconcileSucceeded(ctx, disruptionController, client.ObjectKeyFromObject(machine))
+			machine = ExpectExists(ctx, env.Client, machine)
+			Expect(machine.StatusConditions().GetCondition(v1alpha5.MachineDrifted)).To(BeNil())
+
+			provisioner.Spec.Requirements = []v1.NodeSelectorRequirement{
+				{
+					Key: v1alpha5.LabelCapacityType, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.CapacityTypeSpot},
+				},
+			}
+			ExpectApplied(ctx, env.Client, provisioner)
+			ExpectReconcileSucceeded(ctx, disruptionController, client.ObjectKeyFromObject(machine))
+			machine = ExpectExists(ctx, env.Client, machine)
+			Expect(machine.StatusConditions().GetCondition(v1alpha5.MachineDrifted).IsTrue()).To(BeTrue())
+		})
+		It("should return drifted if a new node requirement is add to the provisioner", func() {
+			cp.Drifted = false
+			provisioner.Spec.Requirements = []v1.NodeSelectorRequirement{
+				{Key: v1alpha5.LabelCapacityType, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.CapacityTypeOnDemand}},
+				{Key: v1.LabelOSStable, Operator: v1.NodeSelectorOpIn, Values: []string{string(v1.Linux)}},
+			}
+			machine.Spec.Requirements = []v1.NodeSelectorRequirement{
+				{Key: v1alpha5.LabelCapacityType, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.CapacityTypeOnDemand}},
+				{Key: v1.LabelOSStable, Operator: v1.NodeSelectorOpIn, Values: []string{string(v1.Linux)}},
+			}
+			ExpectApplied(ctx, env.Client, provisioner, machine, node)
+			ExpectReconcileSucceeded(ctx, disruptionController, client.ObjectKeyFromObject(machine))
+			machine = ExpectExists(ctx, env.Client, machine)
+			Expect(machine.StatusConditions().GetCondition(v1alpha5.MachineDrifted)).To(BeNil())
+
+			provisioner.Spec.Requirements = []v1.NodeSelectorRequirement{
+				{Key: v1alpha5.LabelCapacityType, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.CapacityTypeOnDemand}},
+				{Key: v1.LabelOSStable, Operator: v1.NodeSelectorOpIn, Values: []string{string(v1.Linux)}},
+				{Key: v1.LabelArchStable, Operator: v1.NodeSelectorOpIn, Values: []string{v1alpha5.ArchitectureAmd64}},
```

Comment:
What if this was the chosen architecture anyway for a Machine? Should this Drift?